File: scripts/kaggle_tags.py
#!/usr/bin/env python3
"""Standardized Kaggle Tagging Taxonomy for Pocketgull.

Defines a curated, high-relevance taxonomy of tags and keywords for Kaggle Models,
Datasets, Kernels, and Utility Scripts to maximize platform discoverability and
usability score (10/10).
"""

from typing import List, Optional

# Core Clinical & Medical Imaging Taxonomy
CLINICAL_TAGS = [
    "healthcare",
    "medical-imaging",
    "dicom",
    "mri",
    "fhir",
    "hipaa-safe-harbor",
    "clinical-cds",
    "biomedical",
    "synthetic"
]

# AI, ML & Deep Learning Framework Taxonomy
AI_ML_TAGS = [
    "gemini",
    "agentic-ai",
    "pytorch",
    "onnx",
    "deep-learning",
    "computer-vision",
    "asymmetric-loss",
    "group-kfold",
    "socratic-reasoning"
]

# Project & Benchmark Taxonomy
PROJECT_TAGS = [
    "pocketgull",
    "med-skeptic",
    "rsna",
    "physionet"
]

# Combined master taxonomy
ALL_TAGS = sorted(list(set(CLINICAL_TAGS + AI_ML_TAGS + PROJECT_TAGS)))


def get_standard_tags(
    category: str = "all",
    extra_tags: Optional[List[str]] = None
) -> List[str]:
    """Returns a clean, deduplicated list of standard Kaggle tags.

    Args:
        category: Tag domain category ('all', 'clinical', 'ai_ml', 'project').
        extra_tags: Additional custom tags to append.

    Returns:
        List of formatted string tags.
    """
    if category == "clinical":
        base = list(CLINICAL_TAGS)
    elif category == "ai_ml":
        base = list(AI_ML_TAGS)
    elif category == "project":
        base = list(PROJECT_TAGS)
    else:
        base = list(ALL_TAGS)

    if extra_tags:
        base.extend(extra_tags)

    # Clean and deduplicate preserving case
    seen = set()
    cleaned = []
    for tag in base:
        tag_str = str(tag).strip().lower().replace(" ", "-")
        if tag_str and tag_str not in seen:
            seen.add(tag_str)
            cleaned.append(tag_str)

    return cleaned


if __name__ == "__main__":
    print(f"[OK] Pocketgull Master Tag Taxonomy ({len(ALL_TAGS)} tags):")
    print(", ".join(ALL_TAGS))

File: scripts/dart/kaggle_tags.dart
/// Standardized Kaggle Tagging Taxonomy for Pocketgull (Dart)
///
/// Defines a curated, high-relevance taxonomy of tags and keywords for Kaggle Models,
/// Datasets, Kernels, and Utility Scripts to maximize discoverability and achieve 10/10 usability.

enum TagCategory { all, clinical, aiMl, project }

const List<String> clinicalTags = [
  'healthcare',
  'medical-imaging',
  'dicom',
  'mri',
  'fhir',
  'hipaa-safe-harbor',
  'clinical-cds',
  'biomedical',
  'synthetic',
];

const List<String> aiMlTags = [
  'gemini',
  'agentic-ai',
  'pytorch',
  'onnx',
  'deep-learning',
  'computer-vision',
  'asymmetric-loss',
  'group-kfold',
  'socratic-reasoning',
];

const List<String> projectTags = [
  'pocketgull',
  'med-skeptic',
  'rsna',
  'physionet',
];

List<String> getStandardTags({
  TagCategory category = TagCategory.all,
  List<String>? extraTags,
}) {
  final baseList = switch (category) {
    TagCategory.clinical => [...clinicalTags],
    TagCategory.aiMl => [...aiMlTags],
    TagCategory.project => [...projectTags],
    TagCategory.all => {
        ...clinicalTags,
        ...aiMlTags,
        ...projectTags,
      }.toList()
      ..sort(),
  };

  if (extraTags != null) {
    baseList.addAll(extraTags);
  }

  final seen = <String>{};
  final cleaned = <String>[];

  for (final raw in baseList) {
    final formatted = raw.trim().toLowerCase().replaceAll(' ', '-');
    if (formatted.isNotEmpty && seen.add(formatted)) {
      cleaned.add(formatted);
    }
  }

  return cleaned;
}

void main() {
  final all = getStandardTags();
  print('\n🏷️  Pocket-Gull Standard Kaggle Tag Taxonomy (${all.length} tags):\n');
  print('  ${all.join(', ')}\n');
}
